src/main/java/edu/yu/cs/artAPI/ArtResource.java

```diff
@@ -41,14 +41,16 @@ public List<Art> getAll(@PathParam("gallery-id") long galleryId) {
     }
     
     @GET
-    @Path("/{gallery-id}/arts/namesearch")
+    @Path("/{gallery-id}/arts")
+    //Optional query param of "name"
     //Gets art of a specific name from a specific gallery
     public List<Art> getByName(@PathParam("gallery-id") long galleryId, @QueryParam("name") String name) {
         return ar.findByGallery(galleryId, name);
     }
 
     @GET
-    @Path("/{gallery-id}/arts/creatorsearch")
+    @Path("/{gallery-id}/arts")
```

Comment:
does this work? having two different methods with the same path?

Reply:
Actually no, I just assumed it did but forgot to test 😓. So multiple query params in one method is what you were referring to, right? If so, just wondering if there was a way to structure a Panache Query such that if we provide it default values it will return all results for that parameter. Meaning, in the case of 
`return list("gallery_id = ?1 and name = ?2", galleryId, name);`
is there a way to pass in a value for name such that the returned list returns all pieces of art that just match the galleryId? This would be great as then I could write the method with the optional query terms in basically one line with something like 
`return list("gallery_id = ?1 and name = ?2 and creator =?3", galleryId, name, creator);`
Where we would assign a default to name and creator that allow this result to ignore them when the user does not provide a specific name or creator/

Reply:
yes, multiple query parameters in method. 

Thats an interesting question - I dont think you can quite get what you are look for. 

The best you could do would be a `LIKE` statement with a sql wildcard `%` . But thats not a panache syntax (or even JPA) - more of a sql thing. 

My recommendation would be to have a good ole `if` statement in the `Resource` class checking for existence of the query param.

Reply:
Refactored the code to use the [query params](https://github.com/alinzer/summer-project-step1/pull/1/commits/dc240768ac1f8e2d20e001536700c49423507bf9), also enabled a query by both name and creator